packages/sdk/src/types/runtime/behaviors/clientAction.ts

```diff
@@ -0,0 +1,57 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Context } from '..';
+import { SetAudioStateOptions } from '../../..';
+import * as Payloads from '../../network/payloads';
+
+// tslint:disable: max-classes-per-file
+
+export abstract class ClientAction {
```

Comment:
I would rename this `ClientActionHandler`.  Actions and handlers are very different concepts in the behavior system, and these are handlers for actions.  Not new types of actions

Reply:
I don't love this name, but "action" seems more accurate than "handler" to my mind. They don't handle anything. They just do something. They're actions. We've already defined "behavior", "action", and "handler". Do we need a fourth term for what these are?

---
In reply to: [316833692](https://github.com/microsoft/mixed-reality-extension-sdk/pull/425#discussion_r316833692) [](ancestors = 316833692)

Reply:
They are handling the action that is firing on the behavior.  To call them an action is not inline with what we currently refer to as an action in the SDK.

Reply:
They are more of a reaction to something.  Maybe refer to them as a `ClientReaction`?

Reply:
How about `TriggeredAction`?

---
In reply to: [316852291](https://github.com/microsoft/mixed-reality-extension-sdk/pull/425#discussion_r316852291) [](ancestors = 316852291)

Reply:
That is fine by me.  I just don't like it being called a ClientAction, as the behavior actions are client actions.  I think vocabulary is important here, and is why I am being a bit picky.

Reply:
I actually really like that name now that I think about it.  Let's go with that.  Very descriptive.